### PR TITLE
Feature/clamav

### PR DESCRIPTION
**Pull request by amitschendel**

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -255,6 +257,42 @@ func main() {
 		defer appProfileReconcilerController.StopController()
 	}
 
+	// Start the ClamAV scanner
```
Comment (slashben) [inline cmd/main.go:260]: Isn't this block should be under a condition that the user enabled ClamAV?

File: pkg/scan/clamav.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	for {
+		select {
+		case result, ok := <-response:
+			if !ok {
+				// The response channel was closed, which means the scan is over.
+				log.Println("Scan completed")
+				return
+			}
+			if result.Status == clamd.RES_FOUND {
+				log.Printf("Virus detected: %s", result.Path)
```
Comment (slashben) [inline pkg/scan/clamav.go:65]: I guess we need to alert here, no?

File: chart/kubecop/templates/deamonset.yaml [hunk 3]
```diff
@@ -116,6 +126,21 @@ spec:
...
+        - name: clamdb
+          mountPath: /var/lib/clamav
+          readOnly: false
+        - name: etc
+          mountPath: /etc/clamav
+          readOnly: true
+        - name: host
+          mountPath: /host
+          readOnly: true
+    {{- end }}
```
Comment (slashben) [inline chart/kubecop/templates/deamonset.yaml:143]: We must have memory / CPU limits

File: chart/kubecop/values.yaml [hunk 4]
```diff
@@ -34,6 +34,12 @@ kubecop:
     enabled: true
     ignoreMounts: "true"
     ignorePrefixes: "/proc,/tmp"
+  clamAV:
+    enabled: false
+    host: "localhost"
+    port: 3310
+    path: "/host"
+    scanInterval: 600s # Scan interval in seconds.
```
Comment (slashben) [inline chart/kubecop/values.yaml:42]: How much time takes a scan? Does ClamAV know if it already scanned file and second scans are faster?

File: cmd/main.go (hunk 1 above)
Comment (amitschendel) [reply cmd/main.go:260]: See the code below, I am just initiating the config of the clamav scanner but I check if the variables are actually set.

File: pkg/scan/clamav.go (hunk 2 above)
Comment (amitschendel) [reply pkg/scan/clamav.go:65]: Yep, this is just a placeholder until we will write an exporter.

File: chart/kubecop/templates/deamonset.yaml (hunk 3 above)
Comment (amitschendel) [reply chart/kubecop/templates/deamonset.yaml:143]: How much do we want?

File: chart/kubecop/values.yaml (hunk 4 above)
Comment (amitschendel) [reply chart/kubecop/values.yaml:42]: Let's raise it to be once an hour.
I don't know to tell if the second scans are faster yet...

**Comment by amitschendel** [comment]
@slashben Pls fix the base image to support arm.
Use [this](https://hub.docker.com/layers/clamav/clamav-debian/1.2.0-6/images/sha256-1eb36150a2fd6de165efb88b5a6611e95862f0c16429f5d935cb87a3de448da1?context=explore)
And let's check that the alert manager alerts are looking good before merging to main.

**Pull request merged**